Framework/DataHandling/inc/MantidDataHandling/LoadNGEM.h

```diff
@@ -0,0 +1,155 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#ifndef MANTID_DATAHANDLING_LOADNGEM_H_
+#define MANTID_DATAHANDLING_LOADNGEM_H_
+
+#include "MantidAPI/IFileLoader.h"
+#include "MantidDataObjects/EventWorkspace.h"
+
+namespace Mantid {
+namespace DataHandling {
+
+static constexpr uint64_t CONTIN_ID_VALUE = 0x4F;
+
+/// Generic event to separate bits.
+struct GenericEvent {
+  uint64_t t0id : 24;      // T0 ID
+  uint64_t reserved2 : 32; // Reserved for non-generics
+  uint64_t contin : 8;     // 0x4F Continuation Code
+  uint64_t reserved1 : 56; // Reserved for non-generics
+  uint64_t id : 8;         // 0x4E Event ID
+};
+
+/// Indicate time 0, the start of a new frame.
+struct T0FrameEvent {
+  uint64_t t0id : 24;       // T0 ID
+  uint64_t eventCount : 32; // Event Count
+  uint64_t contin : 8;      // 0x4F Continuation Code
+  uint64_t totalLoss : 24;  // Total loss count
+  uint64_t eventLoss : 20;  // Event loss count
+  uint64_t frameLoss : 12;  // Frame loss count
+  uint64_t id : 8;          // 0x4E Event ID
+  static constexpr int T0_IDENTIFIER = 0x4E;
+  bool check() const {
+    return id == T0_IDENTIFIER && contin == CONTIN_ID_VALUE;
+  }
+};
+
+/// A detected neutron.
+struct CoincidenceEvent {
+  uint64_t t0id : 24;         // T0 ID
+  uint64_t clusterTimeY : 10; // Integrated time of the cluster on the Y side
+                              // (5ns pixel)
+  uint64_t timeDiffY : 6; // Time lag from first to last detection on Y (5ns)
+  uint64_t clusterTimeX : 10; // Integrated time of the cluster on the X side
+                              // (5ns pixel)
+  uint64_t timeDiffX : 6; // Time lag from first to last detection on X (5ns)
+  uint64_t contin : 8;    // 0x4F Continuation Code
+  uint64_t lastY : 7;     // Y position of pixel detected last
+  uint64_t firstY : 7;    // Y position of pixel detected first
+  uint64_t lastX : 7;     // X position of pixel detected last
+  uint64_t firstX : 7;    // X position of pixel detected first
+  uint64_t timeOfFlight : 28; // Difference between T0 and detection (1ns)
+  uint64_t id : 8;            // 0x47 Event ID.
+
+  uint64_t avgX() const { return (firstX + lastX) / 2; }
+  uint64_t avgY() const { return (firstY + lastY) / 2; }
+  static constexpr int COINCIDENCE_IDENTIFIER = 0x47;
+  bool check() {
+    return id == COINCIDENCE_IDENTIFIER && contin == CONTIN_ID_VALUE;
+  }
+  uint64_t getPixel() const {
+    return avgX() + (avgY() << 7); // Increase Y significance by 7 bits to
+                                   // account for 128x128 grid.
+  }
+};
+
+/// Holds the 128 bit words from the detector.
+struct Split64 {
```

Comment:
Can this be renamed, this name isn't entirely obvious as to what this is for.

Also we have Boost multiprecision if you want 128 bit integers.